### PR TITLE
Add com.github.corna.Vivado

### PR DESCRIPTION
**Pull request by corna**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

-----

Hi all.

This PR supersedes #2120 

This PR adds the [Xilinx Vivado Design Suite](https://www.xilinx.com/downloads) as flatpak package. All the Xilinx material is NOT redistributable and requires a login to be downloaded (and the filling of a "U.S. Government Export Approval" form). For this reason, this repository does not contain any Xilinx code and instead works as a wrapper over an "official" installation. The user is prompted to download the official installation tool, which is then passed to this package and installed within the sandbox (in the `$XDG_DATA_HOME/xilinx-install` folder).

This package works (hopefully) with any version of the Xilix Vivado Design Suite. If multiple versions are installed, the user has to select one after clicking on the application icon.

### Permissions
 - `share=ipc`, `socket=wayland` and `socket=x11` are obvious.
 - `share=network` is required for the correct functionality of docnav (which downloads the documentation from the internet) and is required for licensing (as the license is bound to the MAC address). 
 - `device=all` is required for accessing the hardware devices (USB), which can be programmed through Vivado. For this, the user has to install some udev rules, see below.
 - `filesystem=~/.Xilinx:create`: this folder is used by the Xilinx tools to store some data and the license files.
 - `filesystem=xdg-documents` and `filesystem=xdg-desktop`: the user has to store its project somewhere. I don't know however if it is better to provide just the documents and desktop (better sandbox) or the whole home directory (more user-friendly).
 - `persist=.`: not really needed, but it is better to keep the whole home persistent  so that, if the user saves a project outside the Desktop or Documents directory, it is not lost as soon as the application is closed

### Developer contact

Regarding [the requirements][reqs], I have not contacted Xilinx regarding this package (I don't even know where I should contact them, there does not seem to be an email for non-paying customers).

### Icons

The better-suited icons for this package (to be consistent with an official installation) are the Vivado icon, the Vitis icon and the Xilinx logo. These are in the sub-module [com.github.corna.Vivado.icons](https://github.com/corna/com.github.corna.vivado.icons).

### udev rules

The tools can program the boards through many USB-JTAG adapters, which however require an udev rule to correctly set the permissions.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by corna** [comment]
bot, build com.github.corna.Vivado

**Comment by corna** [comment]
bot, build com.github.corna.Vivado

**Comment by corna** [comment]
One thing that is currently missing is support for model composer, which requires MATLAB. This is blocked by:
 * the absence of a flatpak MATLAB package
 * [a way to run a command in another flatpak package](https://github.com/flatpak/xdg-desktop-portal/issues/283)

Alternatively, we could use `flatpak-spawn` to give access to the host's MATLAB installation (and provide a "fake" `matlab` binary which wraps `flatpak-spawn`), but it is like throwing away all the sandboxing efforts.

**Comment by nedrichards** [comment]
Given the constraints that you mention I don't believe that this application is a good fit for flathub. We're trying to help provide a set of applications that "just work" and that can't be true of this app. However flatpak is open for a reason! Users could look to either build it themselves from a manifest (which is super easy) or you could host a repo somewhere else https://docs.flatpak.org/en/latest/hosting-a-repository.html for people who are a little more clued in as to the hoops that have to be jumped through.

**Comment by corna** [comment]
I disagree:
 * filesystem access: I'd lean towards a more strict sandbox, but if the focus of Flathub is to have something very user-friendly, `filesystem=home` can be easily set.
 * udev rules: same as other applications (#278) already present in Flathub. This can be overcome with a command on the "host" (printed during the installation) until a proper solution is implemented in Flatpak.
 * Model composer does not work (for the lack of MATLAB in Flatpak), but is an optional part of Vivado (which, for example, I never used). As soon as a MATLAB Flatpak package is published, support can be easily added to this one.

The only "real" issue is the installation, which does not "just work" unfortunately. However the user process is not that bad:
 * install this package
 * click on any launcher: a dialog appears with the instructions and the correct download web page is automatically opened on the "host"
 * login, fill the fields and download the installer, then select it in the file chooser (automatically opened)
 * select the required features on the installer

After this, everything works correctly, even better than the "official" experience (high resolution icons, integration with the file manager, support for multiple versions, HiDPI support for terminal apps).

I hope these reasons are sufficient to include this package in Flathub, which may help lower the bar for installing Vivado. Typical installations of this software are not trivial and badly integrated in the system. And having a package outside the Flathub ecosystem would not have the same visibility.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Flatpak package for Xilinx Vivado Design Suite
```
Comment (barthalion) [inline README.md:1]: Most, if not all of this info should be in the appdata. Users are unlikely to ever come here.

Comment (corna) [reply README.md:1]: Ok, I will add it

**Comment by corna** [comment]
I have messed up the PR, apparently once I have force-pushed to the PR branch, the PR can not be opened anymore.

Anyways, I have included the README in the appdata file and fixed some remaining bugs, the new PR can be found here #2278

**Pull request closed without merging**